lib/kube/proxy/auth.go

```diff
@@ -174,6 +174,9 @@ func extractKubeCreds(ctx context.Context, cluster string, clientCfg *rest.Confi
 	if err != nil {
 		return nil, trace.Wrap(err, "failed to generate TLS config from kubeconfig: %v", err)
 	}
+	if tlsConfig == nil {
+		return nil, trace.Errorf("tlsConfig was nil")
```

Comment:
Nit: We don't usually use trace.Errorf directly. Also, can we add more diagnostic information to the error message? For example, look at which cases TLSConfigFor can return a nil tlsConfig and log appropriate fields so the logs can help narrow down the issue if it happens again?
```suggestion
		return nil, trace.BadParameter("failed to generate TLS config from kubeconfig xxxxx") 
```